packages/main/src/TimePickerInternals.ts

```diff
@@ -429,6 +505,62 @@ class TimePickerInternals extends UI5Element {
 		}
 		this.setValue(date);
 	}
+
+	/**
+	 * Shifts hours value with +/- 12 depending on hour value and day period.
+	 *
+	 * @param {number} hours current hours
+	 * @returns {number} shifted hours
+	 */
+	_shiftHours(hours: number) {
+		if (this._period === this.periodsArray[0]) { // AM
+			hours = hours === 12 ? 0 : hours;
+		} else if (this._period === this.periodsArray[1]) { // PM
+			hours = hours === 12 ? hours : hours + 12;
+		}
+		return hours;
+	}
+
+	/**
+	 * Clears the currently existing cooldown period and starts new one if requested.
+	 *
+	 * @param {boolean} startNewCooldown whether to start new cooldown period after clearing previous one
+	 */
+	_resetCooldown(startNewCooldown: boolean) {
+		if (!TYPE_COOLDOWN_DELAY) {
+			return; // if delay is 0, cooldown is disabled
+		}
+
+		if (this._typeCooldownId) {
+			clearTimeout(this._typeCooldownId);
+		}
+		if (startNewCooldown) {
+			this._startCooldown();
+		}
+	}
+
+	/**
+	 * Starts new cooldown period.
+	 */
+	_startCooldown() {
+		if (!TYPE_COOLDOWN_DELAY) {
+			return; // if delay is 0, cooldown is disabled
+		}
+
+		this._typeCooldownId = setTimeout(() => {
+			this._kbdBuffer = "";
+			this._typeCooldownId = undefined;
+			if (this._exactMatch) {
+				this._setExactMatch();
+				this._exactMatch = undefined;
```

Comment:
Set to null, as undefined is used when undefined.  <- If we need it just for the falsy value.
Or do we need it as undefined